diff -Naur boot.img-ramdisk1/init.target.rc boot.img-ramdisk0/init.target.rc
--- boot.img-ramdisk1/init.target.rc	2017-08-29 14:31:36.000000000 +0300
+++ boot.img-ramdisk0/init.target.rc	2017-08-29 15:11:39.681680000 +0300
@@ -109,6 +112,31 @@
    group drmrpc input system
    socket fpce stream 0660 system system
 
+on boot
+    chown system system  /dev/goodix_fp
+    chmod 0644 /dev/goodix_fp
+
+on property:ro.boot.fpsensor=fpc
+    setprop persist.sys.fp.goodix 0
+    setprop ro.hardware.fingerprint fpc
+
+on property:ro.boot.fpsensor=gdx
+    setprop persist.sys.fp.goodix 1
+    setprop ro.hardware.fingerprint gdx
+
+service gx_fpd /system/bin/gx_fpd
+    class late_start
+    user system
+    group system
+    disabled
+    writepid /dev/cpuset/system-background/tasks
+
+on property:persist.sys.fp.goodix=0
+    stop gx_fpd
+
+on property:persist.sys.fp.goodix=1
+    start gx_fpd
+
 # Allow usb charging to be disabled peristently
 on property:persist.usb.chgdisabled=1
     write /sys/class/power_supply/battery/charging_enabled 0
